src/main/java/org/broadinstitute/hellbender/engine/spark/BroadcastJoinReadsWithRefBases.java

```diff
@@ -0,0 +1,44 @@
+package org.broadinstitute.hellbender.engine.spark;
+
+import com.google.cloud.dataflow.sdk.transforms.SerializableFunction;
+import org.apache.spark.SparkContext;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.broadcast.Broadcast;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.ReferenceDataflowSource;
+import org.broadinstitute.hellbender.engine.spark.datasources.ReferenceTwoBitSource;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.reference.ReferenceBases;
+import scala.Tuple2;
+
+/**
+ * Joins an RDD of GATKReads to reference data using a broadcast strategy.
+ *
+ * The ReferenceDataflowSource is broadcast using Spark's Broadcast variable mechanism.  The reads are then mapped
+ * over and a refrence query is executed on each read.  This makes sense for ReferenceDataflowSource implementations
+ * that contain the reference data in memory (e.g., ReferenceTwoBitSource), but will likely be much slower for
+ * implementations that have to query other resources for the reference sequences.
+ */
+public class BroadcastJoinReadsWithRefBases {
+    public static JavaPairRDD<GATKRead, ReferenceBases> addBases(final ReferenceDataflowSource referenceDataflowSource,
+                                                                 final JavaRDD<GATKRead> reads) {
```

Comment:
My question is whether this method should be implemented in terms of the other one, by mapping the `RDD<GATKRead>` to a `RDD<GATKRead, Void>`.